### PR TITLE
Adapt KYC verification to v2 of the KYC service

### PR DESCRIPTION
**Pull request by aeddaqqa**
get the nonce and sign it with the privateKey for the new KYC backend implementation
send the nonce signed alongside with the publicKey to the KYC backend to get the access token for Sumsub API

### CODE REVIEW

File: src/store/modules/accounts/accounts.ts [hunk 1]
```diff
@@ -168,9 +168,17 @@ const accounts_module: Module<AccountsState, RootState> = {
...
         async updateKycStatus({ state, rootState }) {
-            const wallet = rootState.activeWallet
-            if (!wallet) return
-            state.kycStatus = await checkVerificationStatus('0x' + wallet.ethAddress)
+            if (!rootState.activeWallet || rootState.activeWallet.type === 'ledger') return null
+            const wallet = rootState.activeWallet as SingletonWallet | MnemonicWallet
+            try {
+                state.kycStatus = await checkVerificationStatus(
+                    wallet.ethKey,
+                    //@ts-ignore
```
Comment (Chriserus) [inline src/store/modules/accounts/accounts.ts:176]: Why is ts-ignore here? Can't we do it properly without ignoring?

File: src/store/modules/accounts/accounts.ts [hunk 2]
```diff
@@ -168,9 +168,17 @@ const accounts_module: Module<AccountsState, RootState> = {
...
+                    rootState.Network.selectedNetwork.name.toLowerCase()
+                )
+            } catch (e) {
+                state.kycStatus = false
```
Comment (Chriserus) [inline src/store/modules/accounts/accounts.ts:185]: Maybe also `console.error(e)` in case of an error

File: src/kyc_api.ts [hunk 3]
```diff
@@ -15,25 +19,46 @@ const kyc_api: AxiosInstance = axios.create({
...
     } catch (e) {
         throw e
     }
 }
 
-async function checkVerificationStatus(address: string): Promise<boolean> {
-    let url = `/verified/${address}`
+async function checkVerificationStatus(
+    privateKey: string,
+    activeNetowrkName: string
```
Comment (Chriserus) [inline src/kyc_api.ts]: Typo, `activeNetworkName`

**Review by Chriserus — CHANGES_REQUESTED** [review]

File: src/store/modules/accounts/accounts.ts (hunk 1 above)
Comment (aeddaqqa) [reply src/store/modules/accounts/accounts.ts:176]: cant because the network is not defined yet

**Review by Chriserus — APPROVED** [review]
LGTM now

**Pull request merged**